internal/testing/val/val.go

```diff
@@ -0,0 +1,28 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package val
+
+// Bool helps get bool pointers for test literals
+func Bool(v bool) *bool { return &v }
+
+// Int helps get int pointers for test literals
+func Int(v int) *int { return &v }
+
+// Int64 helps get int64 pointers for test literals
+func Int64(v int64) *int64 { return &v }
+
+// String helps get string pointers for test literals
+func String(v string) *string { return &v }
```

Comment:
Should we rename these to contain `Ptr` or something in them?

Readability-wise, `PtrFromString("foo")` is better than `String("foo")`

Reply:
Actually, we have an issue here. Because the arguments are copies, each time the function gets called we get a different argument. See https://go.dev/play/p/qXs-hnrChcD

Reply:
keeping these short helps them read nicely, ex:
```
	Pkg: &generated.PkgInputSpec{
		Type:      "maven",
		Namespace: val.String("org.apache.logging.log4j"),
		Name:      "log4j-core",
		Version:   val.String("2.8.1"),
		Subpath:   val.String(""),
	},
```
vs
```
	Pkg: &generated.PkgInputSpec{
		Type:      "maven",
		Namespace: val.PtrFromString("org.apache.logging.log4j"),
		Name:      "log4j-core",
		Version:   val.PtrFromString("2.8.1"),
		Subpath:   val.PtrFromString(""),
	},
```
On another project, I used a lot of go-github, which does this:
https://github.com/google/go-github/blob/v50.1.0/github/github.go#L1499-L1513
ex: https://github.com/google/go-github/blob/v50.1.0/github/actions_artifacts_test.go#L204-L210

Maybe it makes sense to change the package name from `val` to `ptrfrom` ?

Reply:
I think since this is for testing only it is ok to use like this. I was more worried about the pointer change when you call it twice, but that has no impact if this is used only in testing code.

`ptrform` as package name looks good to me, thank you

Reply:
Ah, yep. Added a package comment as well to clarify, and updated the name.